### DISCUSSION TITLE
support for multiple data source in one graph?

### DISCUSSION THREAD

**Discussion by vincentff7** [grafana/grafana#50778] [Dashboards, Panels & Visualizations]
when building SQL  like ' A join B'.
A is from datasource1, for example clickhouse.
B is from datasource2, for example rest API from CMDB.
can support for this? or any suggestions?
thx!

**Comment by ryantxu** [comment]
Have you tried the "-- Mixed --" datasource?  That will let you select set the datasource for each query independently.  The docs are not great for it :(   https://grafana.com/docs/grafana/latest/datasources/

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
